src/bp-templates/bp-nouveau/js/buddypress-activity.js

```diff
@@ -451,8 +451,17 @@ window.bp = window.bp || {};
 					comment_items.each(
 						function( i, item ) {
 							if ( i < comment_items.length - 4 ) {
-								$( item ).addClass( 'bp-hidden' ).hide();
+								
+								//stop hiding elements if the id from hash url for specific comment matches
+								if( window.location.hash && "#" + $( item ).attr('id') === window.location.hash ){
+
+									//in case it's a reply from comment, show hidden parent elements for it to show
+									$( item ).parents( "li.comment-item" ).show();
+									return false;
```

Comment:
@kent-carlo we need to add display all. Because of return might be it'll not execute `// Prepend a link to display all.` section.